features/foutafhandeling.feature

```diff
@@ -5,141 +5,65 @@ Functionaliteit: Afhandeling van fouten
-
-Rule: alleen gespecificeerde parameters mogen worden gebruikt
+Rule: waarde van parameter is valide
 
-    Scenario: Niet gespecificeerde parameter
-        Als '/adressen?straat=Spui' wordt aangeroepen
+    Abstract Scenario: waarde van één parameter is niet correct
+        Als '<path><query string>' wordt aangeroepen
         Dan bevat de response de volgende velden
         | naam   | waarde                                        |
         | title  | Een of meerdere parameters zijn niet correct. |
         | status | 400                                           |
         | code   | paramsValidation                              |
-        | detail | Bad request.                                  |
+        | detail | <detail>                                      |
         En bevat de response de volgende invalidParams
-        | name   | code         | reason                      |
-        | straat | unknownParam | Parameter is niet verwacht. |
-
-Rule: minimaal één optionele parameter is opgegeven
-
-    Abstract Scenario: Er zijn geen parameters opgegeven
-        Als '<path>' wordt aangeroepen
-        Dan bevat de response de volgende velden
-        | naam   | waarde                                          |
-        | title  | Ten minste één parameter moet worden opgegeven. |
-        | status | 400                                             |
-        | code   | paramsRequired                                  |
-        | detail | <error detail>                                  |
-        En bevat de response geen invalidParams
+        | name        | code   | reason                |
+        | <parameter> | <code> | <reason omschrijving> |
 
         Voorbeelden:
-        | path                   | error detail                                                                                                                                          |
-        | /adressen              | Precies 1 parameter van pandIdentificatie, adresseerbaarObjectIdentificatie, zoekresultaatIdentificatie, postcode of huisnummer moet worden opgegeven |
-        | /adresseerbareobjecten | Precies 1 parameter van nummeraanduidingIdentificatie, pandIdentificatie, pandIdentificaties of bbox moet worden opgegeven                            |
-        | /panden                | Precies 1 parameter van adresseerbaarObjectIdentificatie, nummeraanduidingIdentificatie, locatie of bbox moet worden opgegeven                        |
+        | path      | query string                                        | detail                                                                                           | parameter | code             | reason omschrijving                   |
+        | /panden   | ?locatie=abc,def                                    | Ongeldige waarde [abc,def] opgegeven voor parameter [locatie]                                    | locatie   | number           | Waarde is geen geldig decimaal getal. |
```

Comment:
is het handig in de foutmelding de naam van de parameter op te nemen? Bijvoorbeeld "bbox" is wel duidelijk voor de client developer, maar niet voor een eindgebruiker. Dat betekent dat de client applicatie de foutmelding niet door kan zetten naar de eindgebruiker. Was dat niet wel de bedoeling?
Zelfde geldt voor de rechte haken om de waarde en de parameternaam. Begrijpt een eindgebruiker wat die rechte haken betekenen? Moeten dat niet liever aanhalingstekens zijn?

Reply:
N.a.v. ons overleg heb ik de details verwijderd. Een aantal foutmeldingen zijn nu al gebaseerd op de HC common foutafhandeling.feature maar zijn hier specifiek voor het zoeken met locatie en bounding box opgenomen. Details kunnen laten middels #327 evt. worden aangepast.
Daarnaast lijkt het mij niet de bedoeling om de provider foutmeldingen te laten genereren die een client applicatie 1-op-1 kan presenteren aan eindgebruikers. Door de foutmeldingen voor eindgebruikers van een client applicatie aan de provider kant te implementeren ontstaat een te strakke koppeling tussen de client implementatie en de provider implementatie. Dit matcht niet met het principe: loosely coupled.
De client heeft al: opgegeven parameter waarden en krijgt in de response de parameter naam, code en reason terug. Een client applicatie kan zelf een tabel bijhouden met de parameters en codes uit de foutafhandeling.feature en foutmeldingen voor eindgebruikers. Met deze tabel is een client applicatie zelf in staat om een foutmelding aan de eindgebruiker te presenteren en indien gewenst aan te passen zonder daarvoor de provider implementatie te moeten aanpassen.